File: trunk/Phanfare.ExternalAPI/ImageInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace Phanfare.ExternalAPI
{
	public class ImageInfo
	{
		public long AlbumID { get; set; }
		public long SectionID { get; set; }
		public long ImageID { get; set; }
		public string FileName { get; set; }
		public string Caption { get; set; }
		public DateTime? ImageDate { get; set; }
		public bool IsVideo { get; set; }
		public bool IsHidden { get; set; }
		public ImageRendition[] Renditions { get; set; }

		public static ImageInfo FromXML( XmlElement el )
		{
			ImageInfo image = new ImageInfo();
			image.AlbumID = el.GetAttributeInt64( "album_id" );
			image.SectionID = el.GetAttributeInt64( "section_id" );
			image.ImageID = el.GetAttributeInt64( "image_id" );
			image.FileName = el.GetAttribute( "filename" );
			image.Caption = el.GetAttribute( "caption" );
			string imageDate = el.GetAttribute( "image_date" );
			image.ImageDate = string.IsNullOrEmpty( imageDate ) ? null : ( DateTime? )DateTime.Parse( imageDate );
			image.IsVideo = el.GetAttributeBoolean( "is_video" );
			image.IsHidden = el.GetAttributeBoolean( "hidden" );

			XmlElement renditionsElement = el[ "renditions" ];
			image.Renditions = renditionsElement.ReadList<ImageRendition>( "num_renditions", ImageRendition.FromXML );

			return image;
		}
	}
}

File: trunk/PhanfareJavaAPI/src/com/phanfare/api/ImageInfo.java
package com.phanfare.api;

import java.util.Date;

import org.json.me.JSONArray;
import org.json.me.JSONException;
import org.json.me.JSONObject;

public class ImageInfo {
	public long albumId;
	public long sectionId;
	public long imageId;
	public String fileName;
	public String caption;
	public Date imageDate;
	public boolean isVideo;
	public boolean isHidden;
	public ImageRendition[] renditions;

	public static ImageInfo fromJson(JSONObject object) {
		ImageInfo image = new ImageInfo();
		try {
			image.albumId = object.getLong("album_id");
			image.sectionId = object.getLong("section_id");
			image.imageId = object.getLong("image_id");
			image.fileName = object.getString("filename");
			image.caption = object.getString("caption");
			image.imageDate = Utilities.convertJsonDate(object.getString("image_date"));
			image.isVideo = object.getBoolean("is_video");
			image.isHidden = object.getBoolean("hidden");
			JSONArray renditions = object.getJSONArray("renditions");
			image.renditions = new ImageRendition[renditions.length()];
			for (int n = 0; n < image.renditions.length; n++) {
				image.renditions[n] = ImageRendition.fromJson(renditions.getJSONObject(n));
			}
		} catch (JSONException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return null;
		}
		return image;
	}
}
